web/modules/custom/validation_register_form/validation_register_form.module

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for validation_register_form module.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function validation_register_form_form_alter(&$form, FormStateInterface &$form_state, $form_id) {
+  switch ($form_id) {
+    case 'user_register_form':
+      // Set custom validation function.
+      $form['#validate'][] = 'validation_register_form_phone_validate';
+  }
+
+}
+
+/**
+ * Implements user_register_form_validate().
+ */
+function validation_register_form_phone_validate($form, &$form_state) {
```

Comment:
It is better to use underscore at the start of callback names to make them look different than hook functions.
`function _validation_register_form_phone_validate($form, &$form_state)`